src/backend/distributed/utils/remote_commands.c

```diff
@@ -0,0 +1,195 @@
+/*-------------------------------------------------------------------------
+ *
+ * remote_commands.c
+ *   Helpers to make it easier to execute command on remote nodes.
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "libpq-fe.h"
+
+#include "distributed/connection_management.h"
+#include "distributed/remote_commands.h"
+
+
+/* GUC, determining whether statements sent to remote nodes are logged */
+bool LogRemoteCommands = false;
+
+
+/* simple helpers */
+
+/*
+ * IsResponseOK checks whether the result is a successful one.
+ */
+bool
+IsResponseOK(PGresult *result)
+{
+	ExecStatusType resultStatus = PQresultStatus(result);
+
+	if (resultStatus == PGRES_SINGLE_TUPLE || resultStatus == PGRES_TUPLES_OK ||
+		resultStatus == PGRES_COMMAND_OK)
+	{
+		return true;
+	}
+
+	return false;
+}
+
+
+/*
+ * Clear connection from current activity.
+ *
+ * FIXME: This probably should use PQcancel() if results would require network
+ * IO.
+ */
+void
+ForgetResults(MultiConnection *connection)
```

Comment:
It's pretty similar to what we already do in various places, just making the issues it has apparent :(. I'd rather centralize that.